services/api-server/tests/unit/_with_db/test_product.py

```diff
@@ -67,3 +70,43 @@ def _check_key_product_compatibility(request: httpx.Request, **kwargs):
         )
         assert response.status_code == status.HTTP_200_OK
     assert wallet_get_mock.call_count == len(keys)
+
+
+async def test_product_catalog(
+    client: httpx.AsyncClient,
+    mocked_catalog_service_api_base: respx.MockRouter,
+    create_fake_api_keys: Callable[[PositiveInt], AsyncGenerator[ApiKeyInDB, None]],
+) -> None:
+    assert client
+
+    keys: list[ApiKeyInDB] = [key async for key in create_fake_api_keys(2)]
```

Comment:
TIP: here I would clean using param fixtures as I suggested in previous PR. 
Besides keeping it cleaner and more maintainable. In addition it will be more clear the error messages

Reply:
I was reading a bit into the the parametrization for fixtures. But as far as I can tell, if I add 2 parameters to a fixture, that will cause my test to be run independently 2 times. That is not quite what I want here. I want to add two users to the DB associated with two different products and make sure when I log in as one of them I see in the request to the catalog/webserver the product associated with that user. Does that make sense?